private/smithy-rpcv2-cbor/src/protocols/Rpcv2cbor.ts

```diff
@@ -0,0 +1,1227 @@
+    (acc: Record<string, string[]>, [key, value]: [string, any]) => {
+      if (value !== null) {
+        acc[key as string] = _json(value);
+      } else {
+        acc[key as string] = null as any;
+      }
+      return acc;
+    },
+    {} as Record<string, string[]>
+  );
+};
+
+/**
+ * deserializeRpcv2cborSparseStructMap
+ */
+const de_SparseStructMap = (output: any, context: __SerdeContext): Record<string, GreetingStruct> => {
+  return Object.entries(output).reduce(
+    (acc: Record<string, GreetingStruct>, [key, value]: [string, any]) => {
+      if (value !== null) {
+        acc[key as string] = _json(value);
+      } else {
+        acc[key as string] = null as any;
+      }
+      return acc;
+    },
+    {} as Record<string, GreetingStruct>
+  );
+};
+
+// de_StructureList omitted.
+
+// de_StructureListMember omitted.
+
+// de_TestStringList omitted.
+
+// de_TestStringMap omitted.
+
+/**
+ * deserializeRpcv2cborBlobList
+ */
+const de_BlobList = (output: any, context: __SerdeContext): Uint8Array[] => {
+  const collection = (output || []).filter((e: any) => e != null);
+  return collection;
+};
+
+// de_BooleanList omitted.
+
+// de_FooEnumList omitted.
+
+// de_GreetingStruct omitted.
+
+// de_IntegerEnumList omitted.
+
+// de_IntegerList omitted.
+
+// de_NestedStringList omitted.
+
+/**
+ * deserializeRpcv2cborSparseStringList
+ */
+const de_SparseStringList = (output: any, context: __SerdeContext): string[] => {
+  const collection = (output || []).map((entry: any) => {
+    if (entry === null) {
+      return null as any;
+    }
+    return __expectString(entry) as any;
+  });
+  return collection;
+};
+
+/**
+ * deserializeRpcv2cborSparseStringMap
+ */
+const de_SparseStringMap = (output: any, context: __SerdeContext): Record<string, string> => {
+  return Object.entries(output).reduce(
+    (acc: Record<string, string>, [key, value]: [string, any]) => {
+      if (value !== null) {
+        acc[key as string] = __expectString(value) as any;
+      } else {
+        acc[key as string] = null as any;
+      }
+      return acc;
+    },
+    {} as Record<string, string>
+  );
+};
+
+// de_StringList omitted.
+
+// de_StringSet omitted.
+
+/**
+ * deserializeRpcv2cborTimestampList
+ */
+const de_TimestampList = (output: any, context: __SerdeContext): Date[] => {
+  const collection = (output || [])
+    .filter((e: any) => e != null)
+    .map((entry: any) => {
+      return __expectNonNull(__parseEpochTimestamp(entry));
+    });
+  return collection;
+};
+
+const deserializeMetadata = (output: __HttpResponse): __ResponseMetadata => ({
+  httpStatusCode: output.statusCode,
+  requestId:
+    output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
+  extendedRequestId: output.headers["x-amz-id-2"],
+  cfId: output.headers["x-amz-cf-id"],
+});
+
+const throwDefaultError = withBaseException(__BaseException);
+const buildHttpRpcRequest = async (
```

Comment:
Could this be part of a library or is it custom?

Reply:
This can be moved to the `@smithy/core/cbor` submodule.

Reply:
Looking again, `buildHttpRpcRequest` uses [util-body-length-browser](https://github.com/smithy-lang/smithy-typescript/pull/1280/files#diff-ef0fad4ac8f2e42e19d601ca95685947a35dfa2980fb6c4d2ada6b95bde4d03fR4) directly - should this be provided through __SerdeContext depending on the runtime?

Reply:
since the body for CBOR is always Uint8Array, undefined, or streaming, the browser implementation of calculateBodyLength is sufficient for Node.js as well.